File: CS/CustomRecurrenceFormDescendantSample/MyAppointmentRecurrenceForm.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CustomRecurrenceFormDescendantSample
{
    public class MyAppointmentRecurrenceForm : AppointmentRecurrenceForm
    {
        public MyAppointmentRecurrenceForm(Appointment pattern,
            FirstDayOfWeek firstDayOfWeek, AppointmentFormControllerBase controller,
            RecurrenceType type) : base(pattern, firstDayOfWeek, controller)
        {
            base.dailyRecurrenceControl1.Enabled = false;
            base.weeklyRecurrenceControl1.Enabled = false;
            base.monthlyRecurrenceControl1.Enabled = false;
            base.yearlyRecurrenceControl1.Enabled = false;
            switch (type) {
                case RecurrenceType.Daily:
                    base.dailyRecurrenceControl1.Enabled = true;
                    break;
                // Code for other RecurrenceType cases if required.
            }

            base.grpRecurrencePattern.Text =
                "You can edit only daily recurring appointments";
            base.Text = "Appointment Recurrence: Modified Form";
        }
    }

}

File: VB/CustomRecurrenceFormDescendantSample/MyAppointmentRecurrenceForm.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.UI

Namespace CustomRecurrenceFormDescendantSample

    Public Class MyAppointmentRecurrenceForm
        Inherits AppointmentRecurrenceForm

        Public Sub New(ByVal pattern As Appointment, ByVal firstDayOfWeek As FirstDayOfWeek, ByVal controller As AppointmentFormControllerBase, ByVal type As RecurrenceType)
            MyBase.New(pattern, firstDayOfWeek, controller)
            dailyRecurrenceControl1.Enabled = False
            weeklyRecurrenceControl1.Enabled = False
            monthlyRecurrenceControl1.Enabled = False
            yearlyRecurrenceControl1.Enabled = False
            Select Case type
                Case RecurrenceType.Daily
                    dailyRecurrenceControl1.Enabled = True
            ' Code for other RecurrenceType cases if required.
            End Select

            grpRecurrencePattern.Text = "You can edit only daily recurring appointments"
            Text = "Appointment Recurrence: Modified Form"
        End Sub
    End Class
End Namespace
